the-experts-BE/app.js

```diff
@@ -33,6 +39,47 @@ app.post("/addContractor", function(req, res, next) {
   );
 });
 
+app.post("/addReview", function(req, res, next) {
+  queries.addReview(req.body, (err, dataResponse) => {
+    if (err) next(err);
+    res.json(dataResponse);
+  });
+});
+
+app.post("/login", function(req, res, next) {
+  queries.getUser(req.body.email, (err, dataResponse) => {
+    if (err) next(err);
+    comparePasswords(req.body.password,dataResponse.password,(error,result)=>{
+      if(error) next(error);
+      if(!result) res.json(null);
+      const jwt=sign(req.body.email,SECRET);
+      res.cookie('jwt',jwt);
+    res.json(jwt);
+  });
+});
+});
+
+app.post("/signup", function(req, res, next) {
+  let body = {
+    first_name: req.body.first_name,
+    last_name: req.body.last_name,
+    email: req.body.email,
+    user_password: req.body.user_password
+  };
+  let jwt = req.cookies.jwt;
+  if (cookie) res.json(jwt);
```

Comment:
What is `cookie` here? I can't see it being assigned anywhere?

Also you are just sending `jwt` back to FE, then carrying on with `hashPassword()`. Don't you want to check first of all if the user is already signed up, but checking if their username already exists in DB?